### DISCUSSION TITLE
Internal memory

### DISCUSSION THREAD

**Discussion by maverick-2009** [selfcustody/krux#266] [Q&A] [answered]
Is possible remove internal memory from MaixAmigo?
I’d like that my Krux have access only sd card.

**Comment by odudex** [comment, marked as answer]
This is not possible, the firmware boots from internal flash memory. To run stateless OR save settings and encrypted mnemonics on Krux's internal flash is optional. If you believe it's safer to run a signer booting an OS from a removable flash(SD card), there are other options out there 😁

**Reply by maverick-2009** [reply]
Tkz for the answer. 
When I deleted a file from internal memory, that file are permanently deleted or exists a way to recover that? I know if I do it in SD Card, that file can be recovered, but I know that I can destroy a SDcard.

**Reply by odudex** [reply]
Yes it is possible to recover, not as easy as from a SD card, but it is. To make recovery impossible we now have a "wipe device" feature on beta version. It will clean every bit of flash.
The safety of your stored mnemonic relies on encryption, not on the accessibility.
So remembering:
- you have the OPTION to store encrypted mnemonic on internal flash
- you have the OPTION to store encrypted mnemonic on removable flash (SD)
- you have the OPTION to don't store encrypted mnemonic at all.

If you stored an encrypted mnemonic on device's flash and is afraid your encryption key is weak or has leaked AND someone who may know it have physical access to your device AND is able to extract information from the flash, then you can install beta version and perform a full wipe.
